mock/mockimpl/kvservice.go

```diff
@@ -143,10 +144,14 @@ func newKvService(parent *clusterNodeInst, opts newKvServiceOptions) (*kvService
 			LostClientHandler: svc.handleLostMemdClient,
 			PacketHandler:     svc.handleMemdPacket,
 		},
+		ListenPort: *opts.NodeOptions.ListenPort,
 	})
 	if err != nil {
 		return nil, err
 	}
+	if *opts.NodeOptions.ListenPort != 0 {
```

Comment:
This should be controlled from the callee.

Reply:
Sorry, I don't quite understand what you mean here. The ListenPort needs to be incremented before the TLS KV service is started later in the function and if listen port is 0 the TLS service will be listening on port 1 without this check.